site/pages/api/delete-disruption.api.ts

```diff
@@ -0,0 +1,42 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import { getPtSituationElementFromDraft } from "./publish.api";
+import { ERROR_PATH } from "../../constants";
+import { deletePublishedDisruption, getDisruptionById } from "../../data/dynamo";
+import { redirectTo, redirectToError } from "../../utils/apiUtils";
+import logger from "../../utils/logger";
+
+const deleteDisruption = async (req: NextApiRequest, res: NextApiResponse): Promise<void> => {
+    try {
+        const { query } = req;
+
+        const id = query?.id as string;
+
+        if (!id) {
+            throw new Error("Insufficient data provided for deleting a disruption");
+        }
+
+        const disruption = await getDisruptionById(id);
+
+        if (!disruption) {
+            logger.error(`Disruption ${id} not found to delete`);
+            redirectTo(res, ERROR_PATH);
+            return;
+        }
+
+        await deletePublishedDisruption(getPtSituationElementFromDraft(disruption), id);
+
+        redirectTo(res, "/dashboard");
+        return;
+    } catch (e) {
+        if (e instanceof Error) {
```

Comment:
Could potentially be useful to change the code so we have access to the id, and then we can log it out.

Reply:
added it to the error above felt it was not appropriate here